robot/impl/local_robot.go

```diff
@@ -982,6 +987,24 @@ func (r *localRobot) DiscoverComponents(ctx context.Context, qs []resource.Disco
 	return discoveries, nil
 }
 
+// discoverRobotInternals is used to discover parts of the robot that are not in the resource graph
+// It accepts a query and should return the Discovery Results object along with an ok value.
+func (r *localRobot) discoverRobotInternals(query resource.DiscoveryQuery) (interface{}, bool) {
+	switch {
+	// these strings are hardcoded because their existence would be misleading anywhere outside of this function
+	case query.API.String() == "rdk:builtin:module-manager" && query.Model.String() == "rdk:builtin:module-manager":
```

Comment:
where is this API/Model defined?

Reply:
That's the hacky part of this API usage -- there is no API/Model for the module manager -- it exists outside the resource graph. Even if we put it inside the resource graph, the `DiscoveryFunc func(ctx context.Context, logger logging.Logger) (interface{}, error)`  does not accept an instance of the resource so we would need a global modmanager to read the resources from.

Reply:
That is why I put the comment above that equality match. Defining those constants outside of this func could lead to confusion as people assume that if a `modmanager.API` and `modmanager.Model` is defined, then surely it is possible to use other resource-related functions.

Reply:
Are you going to hardcode the same strings on the `app` side? I could see an argument for `const`ing these and importing them from `app` to avoid divergence, but I realize that's kind of a pain. The comment above seems fine if `const`ing sounds like too much work or not feasible for some reason I don't understand.

Reply:
>  could see an argument for `const`ing these and importing them from app to avoid divergence,

It will be called from the app frontend svelte code so it will almost certainly have to be written out twice.